src/Illuminate/Auth/Access/Admission.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Illuminate\Auth\Access;
+
+class Admission
+{
+    /**
+     * Indicates whether admission was allowed.
+     *
+     * @var bool
+     */
+    protected $allowed;
+
+    /**
+     * The reason admission was allowed/denied.
+     *
+     * @var string|null
+     */
+    protected $reason;
+
+    /**
+     * Create a new admission instance.
+     *
+     * @return void
+     */
+    protected function __construct()
+    {
+        //
```

Comment:
we need a comment here perhapse pointing out this method is protected to stop people sending PRs to remove a method they thought was useless